WarehouseManagement.Core/Services/EntryService.cs

```diff
@@ -18,18 +18,63 @@ public EntryService(IRepository repository)
         this.repository = repository;
     }
 
-    public async Task CreateAsync(EntryFormDto model, string userId)
+    public async Task CreateAsync(ICollection<EntryFormDto> model, string userId)
     {
-        var entry = new Entry()
+        foreach (var entry in model)
         {
-            Pallets = model.Pallets,
-            Packages = model.Packages,
-            Pieces = model.Pieces,
-            CreatedAt = DateTime.UtcNow,
-            CreatedByUserId = userId
-        };
+            if (HasExactlyOneTypeSet(entry) == false)
+            {
+                throw new ArgumentException(EntryCanHaveOnlyOneTypeSet);
+            }
+        }
+
+        foreach (var entry in model)
+        {
+            Entry newEntry;
+
+            if (entry.Pallets > 0)
+            {
+                newEntry = new Entry()
+                {
+                    Pallets = entry.Pallets,
+                    Packages = 0,
+                    Pieces = 0,
+                    CreatedAt = DateTime.UtcNow,
+                    CreatedByUserId = userId,
+                    DeliveryId = entry.DeliveryId,
+                    ZoneId = entry.ZoneId,
+                };
+            }
+            else if (entry.Packages > 0)
+            {
+                newEntry = new Entry()
+                {
+                    Pallets = 0,
+                    Packages = entry.Packages,
+                    Pieces = 0,
+                    CreatedAt = DateTime.UtcNow,
+                    CreatedByUserId = userId,
+                    DeliveryId = entry.DeliveryId,
+                    ZoneId = entry.ZoneId,
+                };
+            }
+            else
+            {
+                newEntry = new Entry()
+                {
+                    Pallets = 0,
+                    Packages = 0,
+                    Pieces = entry.Pieces,
+                    CreatedAt = DateTime.UtcNow,
+                    CreatedByUserId = userId,
+                    DeliveryId = entry.DeliveryId,
+                    ZoneId = entry.ZoneId,
+                };
+            }
```

Comment:
lets get this into its own private method, the create method will look much cleaner this way.

Reply:
done